src/main/java/com/nekomaster1000/infernalexp/events/MiscEvents.java

```diff
@@ -214,6 +222,85 @@ public void onRightClickItem(PlayerInteractEvent.RightClickItem event) {
+                    world.playSound(null, ((BlockRayTraceResult)rayTraceResult).getPos(), IESoundEvents.QUARTZ_GLASS_TYPE.getHitSound(), SoundCategory.BLOCKS, 1.0F, 1.0F);
+                }
+                if ((world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() instanceof GlassBlock
+                ||  world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() instanceof PaneBlock)
+                    && !((world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() == IEBlocks.QUARTZ_GLASS.get())
+                    || (world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() == IEBlocks.QUARTZ_GLASS_PANE.get())
+                    )) {
+                    world.destroyBlock(((BlockRayTraceResult)rayTraceResult).getPos(), false);
+                    ((ThrowableBrickEntity) entity).shoot(entity.getMotion().x, entity.getMotion().y, entity.getMotion().z, 0.3F, 1);
+                } else {
+                    ItemEntity itemEntity = new ItemEntity(world, entity.getPosX(), entity.getPosY(), entity.getPosZ(), Items.BRICK.getDefaultInstance());
+                    world.addEntity(itemEntity);
+                    entity.remove();
+                }
+            }
+        }
+        if (entity instanceof ThrowableNetherBrickEntity) {
+            if (rayTraceResult.getType() == RayTraceResult.Type.BLOCK) {
+                if (world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() == IEBlocks.QUARTZ_GLASS.get()
+                    || world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() == IEBlocks.QUARTZ_GLASS_PANE.get()) {
+                    world.playSound(null, ((BlockRayTraceResult)rayTraceResult).getPos(), IESoundEvents.QUARTZ_GLASS_TYPE.getHitSound(), SoundCategory.BLOCKS, 1.0F, 1.0F);
+                }
+                if ((world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() instanceof GlassBlock
+                    ||  world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() instanceof PaneBlock)
+                    && !((world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() == IEBlocks.QUARTZ_GLASS.get())
+                    || (world.getBlockState(((BlockRayTraceResult)rayTraceResult).getPos()).getBlock() == IEBlocks.QUARTZ_GLASS_PANE.get())
+                    )) {
+                    world.destroyBlock(((BlockRayTraceResult)rayTraceResult).getPos(), false);
+                    ((ThrowableNetherBrickEntity) entity).shoot(entity.getMotion().x, entity.getMotion().y, entity.getMotion().z, 0.1F, 1);
+                } else {
+                    ItemEntity itemEntity = new ItemEntity(world, entity.getPosX(), entity.getPosY(), entity.getPosZ(), Items.NETHER_BRICK.getDefaultInstance());
+                    world.addEntity(itemEntity);
+                    entity.remove();
+                }
+            }
```

Comment:
I'm pretty sure, since these are our own entities, this code can be done within the ThrowBrickEntity classes rather than through subscribing to an event. It would be a much more efficient way to handle it if it's possible. I'd look through the ProjectileEntity methods and see if one exists that would work for this.